crates/router/src/core/payments/helpers.rs

```diff
@@ -4540,9 +4541,29 @@ pub async fn get_additional_payment_data(
                 google_pay: None,
             })),
         },
-        domain::PaymentMethodData::PayLater(_) => Ok(Some(
-            api_models::payments::AdditionalPaymentData::PayLater { klarna_sdk: None },
-        )),
+        // domain::PaymentMethodData::PayLater(_) => Ok(Some(
+        //     api_models::payments::AdditionalPaymentData::PayLater { klarna_sdk: None },
+        // )),
+        domain::PaymentMethodData::PayLater(pay_later) => match pay_later {
```

Comment:
we dont need `match pay_later` if outcome is always the same

Reply:
fixed